### ISSUE TITLE
Fix `quiet` option to suppress `report*` warning reports

### ISSUE THREAD

**Issue by astrochemx** [stylelint/stylelint#9375]
### What minimal example or steps are needed to reproduce the bug?

```css
/* stylelint-disable selector-class-pattern -- Ignore */

/* stylelint-disable-next-line selector-id-pattern -- Ignore */
#test__id {
  margin-top: 0.25rem;
}

:root {
  /* stylelint-disable-next-line custom-property-pattern -- Ignore */
  --_test-bg: linear-gradient(transparent, transparent);
}

@media (width >=50em) {
  .lg\:gap-2 {
    gap: 0.5rem;
  }

  .lg\:gap-4 {
    gap: 1rem;
  }

  .lg\:gap-8 {
    gap: 2rem;
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```js
// stylelint.config.mjs
export default {
  defaultSeverity: 'warning',
  extends: ['stylelint-config-standard'],
};
```

### How did you run Stylelint?

PROBLEMATIC, errors are reported:
```shell
stylelint --report-needless-disables --quiet "**/*.css"
```

OK, no errors are reported:
```shell
stylelint --report-needless-disables "**/*.css"
```

[Reproduction Repo](https://github.com/astrochemx/stylelint-issue-repro-1)

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcID0AqABAZwC4E8A2MeAlgHZYC0URGAhgEYGaExhYQBO5YeNGG5ABxpYsMdiTTlyaAJIBzEhxhoUSADokNqTLgLEylavQLkSMAB4V9yjM1YdyRKIOGjxk6fMXtlqjQGJRbAB9YKc0YA00NABbGnY5UnI2AXg0AAYAOgAmAFYfGIBuDQBfDQ14dggILAiotG1sfEJSCipaBhhTCytSZTAAV2wIGMEqgTFcFxExCSlZBSUVdQkPYKCKOjk063jyOXYaKhgyAAosQ5IMIR8yABo0C5orm5OsAEpikjLNEgABGIwKg0NCnADuTiwAAs0AA+AC8aFy6RgMXedVWmTwcjUangchoAnI2Qx0WiBNSGUy+VRX2iPx+IDuIAAZkQCAA5GiAhAgCzcgQETJgPhM8AQEhsuS8iwCDi1WAsmgDPC1SKrRXK1UAZRgADcxERcGkAORg+IkUhyE13eo9E5QDBpADaJqaelaXAlUvI2GeUHiUBNAF1bd9CmLIJKiHIAGIcOJYXnulpkYXemOZGIAKwwYtgAjziHV0TUIBT+iwZbSZZ4GzLYdL5d0qYoUZ9fpIAfYUGraFrwhg2DLpQ0IBKQA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "^17.14.0",
  "stylelint-config-standard": "^40.0.0"
}
```

### What did you expect to happen?

Expected both runs of Stylelint to report no errors

### What actually happened?

✅ Stylelint with `--report-needless-disables` option set reports no errors:

```pwsh
PS C:\UserFiles\GHub\stylelint-issue-repro> pnpm stylelint --report-needless-disables "**/*.css"
```

❌ Stylelint with `--report-needless-disables` and `--quiet` options set reports errors:

```pwsh
PS C:\UserFiles\GHub\stylelint-issue-repro> pnpm stylelint --report-needless-disables --quiet "**/*.css"

test.css
  1:1  ⚠  Needless disable for "selector-class-pattern"   --report-needless-disables
  3:1  ⚠  Needless disable for "selector-id-pattern"      --report-needless-disables
  9:3  ⚠  Needless disable for "custom-property-pattern"  --report-needless-disables

⚠ 3 problems (0 errors, 3 warnings)
```

### Do you have a proposal to fix the bug?

Unfortunately, I do not have a proposal to fix the bug.
However, there is maybe a related issue back from 2024: https://github.com/stylelint/stylelint/issues/7843

**Comment by ybiquitous** [comment]
@astrochemx Thank you for the report. I've also confirmed the bug in [this minimal reproducible demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcID0AqABAZwC4E8A2MeAlgHZYC0URGAhgEYFoBmEEaKSAOiaprgcTKVq9RuXJoaaWBjAAnIgAcsRCCXZcSIADQgmRAgDkaAWziIYAD1OKCAOjAYMO8Gv0BzBCCuKIcrNIwTDQArngBwNxoaHIwvv4AIjCyCsqqJAROCSIMyfBoANpoWHIhMNpowJgwAG4wCrj5AOQA7jRyJKTuTWgAvmgAutpRMXF+WACSJDU0xFAAypCKMNm0uRj5RSVlFVUYtfVEjWit7Z0k3X2Dw+qj8ViGMDBQmRironmFxaXlldV1DRwzTaHS6PX6QxGsXuAFUSLIIMsoO91ptvjs-vsAUcgScQedLhCbtEAI4hIgwLD5bblKFhT7AXo3XoAbm4LkgJA8ADE-CYaFgvNh8IRSFgHG4iO47CYAFbOXSwRTORCRW6cEDCgRijX5DUARgA7HZ9QAWOwABg13F67N6QA):

```css
/* stylelint-disable foo */
/* stylelint-disable -- a description */
```

```js
export default {
  reportDescriptionlessDisables: [ true, { severity: 'warning' } ],
  reportInvalidScopeDisables: [ true, { severity: 'warning' } ],
  reportNeedlessDisables: [ true, { severity: 'warning' } ],
  reportUnscopedDisables: [ true, { severity: 'warning' } ],
  quiet: true,
  rules: {},
};
```

```sh-session
$ npx stylelint test.css

test.css
  1:1  ⚠  Needless disable for "foo"                  --report-needless-disables
  1:1  ⚠  Rule "foo" isn't enabled                    --report-invalid-scope-disables
  1:1  ⚠  Disable for "foo" is missing a description  --report-descriptionless-disables
  2:1  ⚠  Needless disable for "all"                  --report-needless-disables
  2:1  ⚠  Configuration comment must be scoped        --report-unscoped-disables

⚠ 5 problems (0 errors, 5 warnings)
```

As the reproduction shows, the bug affects all `report*` options, not only `report-needless-disables`.

I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/contributor-guide/rules#fix-a-bug-in-a-rule) in the Contributor guide.
